module/behaviour/skills/DirectWalkController/src/DirectWalkController.cpp

```diff
@@ -72,16 +79,20 @@ namespace module::behaviour::skills {
 
         on<Trigger<MotionCommand>>().then([this](const MotionCommand& command) {
             if (command.type == MotionCommand::Type::DIRECT_COMMAND) {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {26, 11}}));
-                emit(std::make_unique<WalkCommand>(subsumptionId, command.walk_command));
+                update_priority(cfg.direct_walk_priority);
+                emit(std::make_unique<WalkCommand>(subsumption_id, command.walk_command));
             }
             else if (command.type == MotionCommand::Type::STAND_STILL) {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {26, 11}}));
-                emit(std::make_unique<StopCommand>(subsumptionId));
+                update_priority(cfg.direct_walk_priority);
+                emit(std::make_unique<StopCommand>(subsumption_id));
             }
             else {
-                emit(std::make_unique<ActionPriorities>(ActionPriorities{subsumptionId, {0, 0}}));
+                update_priority(0);
```

Comment:
Why is it that every other motion command other than direct or stand still is zero? Is there a situation that we will get another motion command other than these two and they need to be set to zero?

Reply:
If any other motion command is happening, DirectWalkController doesn't handle that and so shouldn't be doing anything (ie no priority on limbs).